src/libsyntax/unicode/char.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright 2012-2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use super::tables::{derived_property, charwidth};
+
+/// Returns whether the specified character satisfies the 'XID_Start'
+/// Unicode property.
+///
+/// 'XID_Start' is a Unicode Derived Property specified in
+/// [UAX #31](http://unicode.org/reports/tr31/#NFKC_Modifications),
+/// mostly similar to ID_Start but modified for closure under NFKx.
+#[unstable(feature = "unicode",
+           reason = "mainly needed for compiler internals")]
+#[inline]
+pub fn is_xid_start(ch: char) -> bool { derived_property::XID_Start(ch) }
+
+/// Returns whether the specified `char` satisfies the 'XID_Continue'
+/// Unicode property.
+///
+/// 'XID_Continue' is a Unicode Derived Property specified in
+/// [UAX #31](http://unicode.org/reports/tr31/#NFKC_Modifications),
+/// mostly similar to 'ID_Continue' but modified for closure under NFKx.
+#[unstable(feature = "unicode",
+           reason = "mainly needed for compiler internals")]
+#[inline]
+pub fn is_xid_continue(ch: char) -> bool { derived_property::XID_Continue(ch) }
+
+/// Returns this character's displayed width in columns, or `None` if it is a
+/// control character other than `'\x00'`.
+///
+/// `is_cjk` determines behavior for characters in the Ambiguous category:
+/// if `is_cjk` is `true`, these are 2 columns wide; otherwise, they are 1.
+/// In CJK contexts, `is_cjk` should be `true`, else it should be `false`.
+/// [Unicode Standard Annex #11](http://www.unicode.org/reports/tr11/)
+/// recommends that these characters be treated as 1 column (i.e.,
+/// `is_cjk` = `false`) if the context cannot be reliably determined.
+#[deprecated(reason = "use the crates.io `unicode-width` library instead",
+             since = "1.0.0")]
+#[unstable(feature = "unicode",
+           reason = "needs expert opinion. is_cjk flag stands out as ugly")]
+pub fn width(ch: char, is_cjk: bool) -> Option<usize> {
+    charwidth::width(ch, is_cjk)
+}
```

Comment:
I think usage of this should be hardwired now to just a width of 1 so this may not need to move here